internal/kafka/internal/config/dynamic_scaling_config_test.go

```diff
@@ -3,324 +3,503 @@ package config
+			name: "should return an error when ClusterWideWorkload is missing",
 			fields: fields{
-				Configuration: map[string]InstanceTypeDynamicScalingConfig{},
+				ClusterWideWorkload: nil,
+				KafkaWorkloadPerInstanceType: map[string]ComputeMachineConfig{
+					"type": {},
+				},
 			},
-			want: map[string]InstanceTypeDynamicScalingConfig{},
+			args: args{
+				cloudProvider: cloudproviders.AWS,
+			},
+			wantErr: true,
 		},
 		{
-			name: "should return a copy of the original configuration map",
+			name: "should return an error when KafkaWorkloadPerInstanceType is missing",
 			fields: fields{
-				Configuration: map[string]InstanceTypeDynamicScalingConfig{
-					"instance-type": {
-						ComputeNodesConfig: &DynamicScalingComputeNodesConfig{
-							MaxComputeNodes: 10,
+				ClusterWideWorkload:          &ComputeMachineConfig{},
+				KafkaWorkloadPerInstanceType: nil,
+			},
+			args: args{
+				cloudProvider: cloudproviders.AWS,
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error when ClusterWideWorkload is invalid when min nodes is less than 3",
+			fields: fields{
+				ClusterWideWorkload: &ComputeMachineConfig{
+					ComputeMachineType: "some-machine",
+					ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+						MinComputeNodes: 1, // This will cause a validation error as min should be 3
+						MaxComputeNodes: 9,
+					},
+				},
+				KafkaWorkloadPerInstanceType: map[string]ComputeMachineConfig{
+					"type": {
+						ComputeMachineType: "machine-type",
+						ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+							MaxComputeNodes: 1,
+							MinComputeNodes: 1,
+						},
+					},
+				},
+			},
+			args: args{
+				cloudProvider: cloudproviders.AWS,
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error when ClusterWideWorkload is invalid when marshalled configurations are missing",
+			fields: fields{
+				ClusterWideWorkload: &ComputeMachineConfig{
+					ComputeMachineType: "",
+					ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+						MinComputeNodes: 0,
+						MaxComputeNodes: 0,
+					},
+				},
+				KafkaWorkloadPerInstanceType: map[string]ComputeMachineConfig{
+					"type": {
+						ComputeMachineType: "machine-type",
+						ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+							MaxComputeNodes: 1,
+							MinComputeNodes: 1,
 						},
 					},
-					"instance-type-2": {
-						ComputeNodesConfig: &DynamicScalingComputeNodesConfig{
+				},
+			},
+			args: args{
+				cloudProvider: cloudproviders.AWS,
+			},
+			wantErr: true,
+		},
+		{
+			name: "should return an error when KafkaWorkloadPerInstanceType is invalid",
+			fields: fields{
+				ClusterWideWorkload: &ComputeMachineConfig{
+					ComputeMachineType: "some-machine",
+					ComputeNodesAutoscaling: &ComputeNodesAutoscalingConfig{
+						MinComputeNodes: 3, // This will cause a validation error as min should be 3
```

Comment:
I don't understand this. `MinComputeNodes` is 3 already

Reply:
hmm.. this doesn't look right. Could be a copy/paste of the config issue. Will relook.

Reply:
Yeap. copy/paste issue of the config from above test cases.